src/framework/uicomponents/qml/MuseScore/UiComponents/internal/GridViewDelegate.qml

```diff
@@ -41,14 +41,17 @@ Item {
     GridView {
         id: gridView
 
-        property int columns: root.columns !== -1 ? root.columns : gridView.count
+        property int columns: root.columns !== -1 ? root.columns : Math.ceil(gridView.count / gridView.rows)
```

Comment:
I think you have a binding loop warning here because columns depend on rows and rows are also column dependent

Reply:
Ah, good point, that will happen if root.rows and root.columns are both -1. However, isn't that a bad situation anyway?

Reply:
What problem did you solve here? Because the old solution seems logical to me

Reply:
if both are -1 then 
columns = gridView.count 
rows = Math.ceil(gridView.count / gridView.count) = 1
in the old version

Reply:
But what if rows = 2 (given) and columns = -1?
In the old version, `columns` would then become just `gridView.count`, but it should be `Math.ceil(gridView.count / gridView.rows)`. (Otherwise, there would be 2x too much cells).
This case is not used anywhere in MuseScore yet, but it seems better to make that work too, in case we need it in the future.

Or am I now mistaken? 🤯

I think I should add a special case for when rows and columns are both -1, so that rows defaults to +1 in that case.